compiler/llvm/clangUtil.cpp

```diff
@@ -2365,6 +2365,8 @@ static std::string generateClangGpuLangArgs() {
       case GpuCodegenType::GPU_CG_AMD_HIP:
         args += "hip";
         break;
+      case GpuCodegenType::GPU_CG_CPU:
```

Comment:
That was my intention first. But I couldn't find a flag that does that in `clang`. The typical way is to use `clang++` instead. I couldn't be certain if that is equivalent to `-x cuda`/`-x hip`, or whether it has other implications. So, I shied away from it.

Reply:
Looks like it's valid to do `-x c++` but "hello world" fails unless I also explicitly pass in `-lstdc++`.

If it's easy to do I think it's worth trying to see if we can get this to work with `-x c++` and `-lstdc++` but if there's any hiccups then move on by filing a public GH issue, documenting this caveat in the technote, and considering it a low-priority feature request to get this mode to do c+++ compilation waiting to see if we get any user feedback or discover that it would be useful for our purposes.